homeassistant/components/ecobee/sensor.py

```diff
@@ -167,3 +156,19 @@ async def async_update(self):
                     continue
                 self._state = item["value"]
                 break
+
+    async def async_added_to_hass(self):
+        """Run when entity is added to hass."""
+        if self.entity_description.key == "temperature":
+            # Tell HA that the entity's native unit is the same
+            # as HA config's temperature unit. Then we will do
+            # the needed conversion and rounding in native_value.
+            self._attr_native_unit_of_measurement = (
+                self.hass.config.units.temperature_unit
+            )
+        else:
```

Comment:
`else` is redundant

Reply:
It isn't returning, so without else would it not always overwrite itself? even if it went through the if block?

Reply:
You can also drop also the content of the else. It will fallback automatically to the entity_description.